server/src/routes/listings.ts

```diff
@@ -1,18 +1,22 @@
 import { Request, Response } from "express";
 import * as express from "express";
 export const listingRoutes = express.Router();
-//import {getAllListing} from "../data"
+import {getAllListings} from "../data"
 import * as distance from "geo-distance"
 
 
 /**Getting listing and sorting from closest to furthest 
- * Send request to /search/location?lat=xxxx&lon=xxx
+ * Client send request to /search/location?lat=xxxx&lon=xxx
 */
 listingRoutes.get('/search/location', async (req: Request, res: Response) => {
     console.log("GET /listings/search/location");
-    /** Send request to /search/location?lat=xxxx&lon=xxx */
-    const location = { lat : req.query.lat, lon : req.query.lon}
-
-    res.json({message: 'successfuly got listings'});
+    const location = { lat : parseFloat(req.query.lat.toString()), lon : req.query.lon.toString()}
+    const listings = await getAllListings();
+    const sortedListings = listings.sort((a,b) => {
+        const aDistance = distance.between(location, {lat: a.address.geolocation[0], lon: a.address.geolocation[1]}).human_readable().distance;
+        const bDistance = distance.between(location, {lat: b.address.geolocation[0], lon: b.address.geolocation[1]}).human_readable().distance;
+        return aDistance - bDistance;
+    });
+    res.json(listings);
```

Comment:
Looks good, was just wondering if this was supposed to return the sortedListings instead of just the listings.

Reply:
It should be returning sorted listings closest to the geolocation that is being sent. If you use postman and add ?lat=xxx&lon=xxx it will do whatevers closest first going further.

Reply:
Yeah when I test it in Postman, I just get the same ordering despite whatever lat and lon I give it. I think you do sort in already and have it in the sortedListings variable, but you return just the listings variable instead.

Reply:
i see what you mean i forgot to change it back LOL. Ill change that thanks for pointing it out